wifiphisher/common/victim.py

```diff
@@ -0,0 +1,189 @@
+                    else:
+                        raise Exception("Error: No such mac address exists in victim dictionary")
+
+
+    """
+    receives a victims ip address and request  as input, finds the corresponding hostname by reading the initial requests file
+    and then accesses the victim dictionary and changes the hostname for the victim with the corresponding ip address
+    :param self: empty Victim instance
+    :type self: Victim
+    :param ip_address: ip address of the victim
+    :type ip_address: str
+    :param url: initial request of the victim
+    :type url: str
+    """
+    def associate_victim_ip_to_hostname(self, ip_address, url):
+        with open(path_to_url_file, "r") as f:
+            for line in f:
+                line = line.rstrip('\n')
+                index = line.find("|")
+                url_check = line[index+1:]
+                url_check = url_check.strip()
+                hostname = line[:index]
+                hostname = hostname.strip()
+                if url_check in url:
+                    Victims_instance = Victims.getInstance()
+                    for key in Victims_instance.victims_dic.keys():
+                        if ip_address == Victims_instance.victims_dic[key].ip_address:
+                            Victims_instance.victims_dic[key].hostname = hostname
+
+
+
+     #updates the ip address of the victim with the corresponding mac address
+    def assign_ip_address_to_victim_object(self, vmac_address, ip_address):
+        Victims_instance = Victims.getInstance()
+        if vmac_address in Victims_instance.victims_dic.keys():
+            Victims_instance.victims_dic[vmac_address].ip_address = ip_address
+        else:
+            raise Exception("Error: No such mac address exists in victim dictionary")
+
+
+
+
+#singleton class which has a dictionary of victims connected to the AP [key:mac, value:Victim object]
+class Victims():
+
+
+    # Instance will be stored here.
+    __instance = None
+
+    @staticmethod
+    def getInstance():
+        """ Static access method. """
+        if Victims.__instance == None:
+            Victims()
+        return Victims.__instance
+
+    def __init__(self):
+        if Victims.__instance != None:
+            raise Exception("Error: This class is a singleton!")
+        else:
+            Victims.__instance = self
+            self.victims_dic = {}
+
+    #add new victim to the dictionary
+    def add_to_victim_dic(self, Victim):
+        self.victims_dic[Victim.vmac_address] = Victim
+
+    #delete victim by Victim object
+    def remove_from_victim_dic(self, Victim):
+        if Victim.vmac_address in self.victims_dic.keys():
+            self.victims_dic.pop(Victim.vmac_address, 0)
+        else:
+            raise Exception("Error: Victim to be deleted Not found")
+
+     #delete victim by victim mac address
+    def remove_from_victim_dic_bymac(self, vmac_address):
```

Comment:
This looks unused. We can remove it.

Reply:
i though about leaving it in case of future versions where we delete victims if they disconnect